test/r_c_shortest_paths_test.cpp

```diff
@@ -199,6 +199,89 @@ class dominance_spptw
 };
 // end data structures for shortest path problem with time windows (spptw)
 
+struct spp_spptw_marked_res_cont {
+    spp_spptw_marked_res_cont(SPPRC_Example_Graph::vertex_descriptor v, int c = 0, int t = 0 ) : cost( c ), time( t ), marked() {
+        marked.insert(v);
+    }
+    spp_spptw_marked_res_cont& operator=( const spp_spptw_marked_res_cont& other )
+    {
+        if( this == &other )
+            return *this;
+        this->~spp_spptw_marked_res_cont();
+        new( this ) spp_spptw_marked_res_cont( other );
```

Comment:
Invoking the destructor and then the constructor is not a good way to write an assignment operator. Perhaps it works here, I haven't checked, but it's not a good practice and it would be better to rewrite it in a more conventional way.

Reply:
@pdimov, I agree. I implemented proper assignment operators in tests whenever possible. Only one has left in https://github.com/boostorg/graph/blob/develop/include/boost/graph/r_c_shortest_paths.hpp#L47. See this is part of released Boost. It was done by the original developer of the feature to allow reassignment of fields marked as const in the struct definition.

I was not changing it deliberately, because it has been already released and it is not causing any bugs as far as I am concerned. It is quite unusual construct to use without a reason.